src/sqlBible/src/SqlBible.ts

```diff
@@ -0,0 +1,292 @@
+import 'reflect-metadata';
+import { createConnection, ConnectionOptions, Connection, getManager, Raw, Between } from 'typeorm';
+
+import { BOOK_DATA } from './data/bibleMeta';
+import {
+    ENTITIES,
+    BibleVersion,
+    BiblePhrase,
+    BibleBook,
+    BibleSection,
+    IBibleReferenceRangeVersion,
+    IBibleReferenceRangeNormalized,
+    IBibleReferenceNormalized,
+    IBibleReferenceVersion,
+    BibleCrossReference
+} from './models';
+import { parsePhraseId, generateNormalizedRefId } from './utils';
+
+export class SqlBible {
+    dbReady: Promise<Connection>;
+    currentVersion?: BibleVersion;
+    currentVersionMetadata?: BibleBook[];
+
+    constructor(dbConfig: ConnectionOptions) {
+        this.dbReady = createConnection({
+            ...dbConfig,
+            entities: ENTITIES,
+            synchronize: true,
+            logging: ['error']
+        });
+    }
+
+    async addBook(book: BibleBook) {
+        await this.dbReady;
+        return await getManager().save(book);
+    }
+
+    async addParagraph(phrases: BiblePhrase[]) {
+        await this.dbReady;
+        const newPhrases = await this.addPhrases(phrases);
+        const section = new BibleSection({
+            phraseStartId: newPhrases[0].id!,
+            phraseEndId: newPhrases[newPhrases.length - 1].id!,
+            level: 0
+        });
+        return getManager().save(section);
+    }
+
+    async addPhrase(phrase: BiblePhrase) {
+        await this.dbReady;
+        await this.preparePhraseForDb(phrase);
+        return await getManager().save(phrase);
+    }
+
+    async addPhrases(phrases: BiblePhrase[]) {
+        await this.dbReady;
+        let verse,
+            phraseNum = 0;
+        for (const phrase of phrases) {
```

Comment:
I prefer `forEach` but maybe that's just personal preference

Reply:
for would be the most generic in this case and in my feeling it's super clean and simple. what would be an added benefit for using the more complex forEach here?